src/snowflake/cli/_plugins/stage/manager.py

```diff
@@ -306,23 +309,116 @@ def put(
         overwrite: bool = False,
         role: Optional[str] = None,
         auto_compress: bool = False,
+        use_dict_cursor: bool = False,
     ) -> SnowflakeCursor:
         """
         This method will take a file path from the user's system and put it into a Snowflake stage,
         which includes its fully qualified name as well as the path within the stage.
         If provided with a role, then temporarily use this role to perform the operation above,
         and switch back to the original role for the next commands to run.
         """
+        if "*" not in str(local_path):
+            local_path = (
+                str(local_path) + "/*" if Path(local_path).is_dir() else str(local_path)
```

Comment:
Shouldnt we do the join using path, and cast the final result to string?

Reply:
Good point, fixed